SICAR/sicar.py

```diff
@@ -276,3 +322,40 @@ def download_country(self, base_folder: str = "Brazil", debug: bool = False):
                 # store log file with failed codes
                 with open(folder + "/failed_codes.txt", "w") as f:
                     print(details, file=f)
+
+    def download_city_csv(
+        self, city_code: str, tries: int = 25, folder: str = "temp", debug: bool = False
+    ):
+        Path(folder).mkdir(parents=True, exist_ok=True)
+        captcha = ""
+
+        while tries > 0:
+            try:
+                captcha = self.__driver._get_captcha(
+                    self._download_captcha(folder=folder)
+                )
+
+                if len(captcha) == 5:
+                    if debug:
+                        print(
+                            "Try {} - Requesting csv file with captcha: {}".format(
+                                tries, captcha
+                            )
+                        )
+
+                    return self._download_csv(
+                        city_code=city_code, captcha=captcha, folder=folder
+                    )
+                else:
+                    if debug:
+                        print("Invalid Captcha: {}".format(captcha))
+
+                    time.sleep(0.75 + random.random() + random.random())
+
+            except Exception:
+                if debug:
+                    print("Try {} - Incorret captcha: {} :-(".format(tries, captcha))
+                tries -= 1
+                time.sleep(1 + random.random() + random.random())
+
+        return False
```

Comment:
`download_city_code` and `download_city_csv` are almost the same code, and the CSV download option is not available in the other methods.

Let's add a parameter to `download_city_code`. I'm expecting something like:

```
def download_city_code(self, city_code: str, mode = "shapefile", tries: int = 25, folder: str = "temp", debug: bool = False):
    ...
    if len(captcha) == 5:
        if debug:
            print(f"Try {tries} - Requesting {mode} file with captcha: {captcha}")

    if mode is "csv":
        return self._download_csv(city_code=city_code, captcha=captcha, folder=folder)
    
    return self._download_shapefile(city_code=city_code, captcha=captcha, folder=folder)
```

So `mode` can be passed to all other methods as well.

If you want to make it even better, I suggest using an `Enum`.

```
from enum import Enum

class Mode(str, Enum):
    SHAPEFILE = "shapefile"
    CSV = "csv"

def download_city_code(self, city_code: str, mode = Mode.SHAPEFILE, tries: int = 25, folder: str = "temp", debug: bool = False):
    ...

    if len(captcha) == 5:
        if debug:
            print(f"Try {tries} - Requesting {mode} file with captcha: {captcha}")

    if mode is Mode.CSV:
        return self._download_csv(city_code=city_code, captcha=captcha, folder=folder)
    
    return self._download_shapefile(city_code=city_code, captcha=captcha, folder=folder)
```

Reply:
That is a great ideia! Thanks for the feedback.
I've done as suggested, please feel free to give me more feedback. I used the name `output_format` to the parameter and the enum, because i thought this name would reveal more about its purpose. What do you think? We can go back to mode if you think it's better! =)

I also updated the `README.md` file, to add an example use case of the csv option. I did not updated the roadmap section, because I didn't know if this PR fully satisfies your original intention with the .csv support.

Again, thank you for the feedback!

Reply:
Thanks, and I appreciate your naming convention.

I've included the CSV option in the other methods and will do some testing before merging it.